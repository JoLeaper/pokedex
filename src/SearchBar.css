.search {
    display: flex;
}

.searchbox {
    display: flex;
    background-color: gray;
    width: 20%;
    height: auto;
    flex-direction: column
}

.resultsBox {
    width: 80%;
    display: grid;
    grid-template-columns: 1fr
}
