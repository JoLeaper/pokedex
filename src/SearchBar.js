import React, { Component } from 'react'
import SortSelector from './SortSelector'
import './SearchBar.css'

export default class SearchBar extends Component {
  render() {
    return (
      <div className='searchbox'>
        Name:
        <input value={this.props.pokemonName} onChange={this.props.handleNameChange}>
        </input>
        <SortSelector
          displayOrder={this.props.displayOrder}
          sortBy={this.props.sortBy}
          handleOrderChange={this.props.handleOrderChange}
          handleSortChange={this.props.handleSortChange} />

        Type:
        <select onChange={this.props.handleTypeChange}>
          {
            this.props.pokemonType.map(type =>
              <option value={type}>{type}</option>
            )
          }
        </select>
                Minimum Attack Of:<input value={this.props.pokemonAttack} onChange={this.props.handleAttackChange}>
        </input>
        <button onClick={this.props.handleClick}>Search</button>
        <button onClick={this.props.handlePageClick} value='next'>Next Page</button>
        <button onClick={this.props.handlePageClick} value='prev'>Previous Page</button>
        <span>
        </span>
      </div>
    )
  }
}
