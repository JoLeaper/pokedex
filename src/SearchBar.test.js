import React from 'react';
import { shallow } from 'enzyme';
import SearchBar from './SearchBar'
const pokemonType = ['', 'normal', 'fire', 'fighting', 'water', 'flying', 'grass', 'poison', 'electric', 'ground', 'psychic', 'rock', 'ice', 'bug', 'dragon', 'ghost', 'dark', 'steel', 'fairy'];

test('renders searchbar', () => {
    const wrapper = shallow(<SearchBar pokemonType={pokemonType} />);
    expect(wrapper).toMatchSnapshot();
});
