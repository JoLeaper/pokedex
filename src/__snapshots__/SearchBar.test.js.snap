// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders searchbar 1`] = `
<div
  className="searchbox"
>
  Name:
  <input />
  <SortSelector />
  Type:
  <select>
    <option
      value=""
    />
    <option
      value="normal"
    >
      normal
    </option>
    <option
      value="fire"
    >
      fire
    </option>
    <option
      value="fighting"
    >
      fighting
    </option>
    <option
      value="water"
    >
      water
    </option>
    <option
      value="flying"
    >
      flying
    </option>
    <option
      value="grass"
    >
      grass
    </option>
    <option
      value="poison"
    >
      poison
    </option>
    <option
      value="electric"
    >
      electric
    </option>
    <option
      value="ground"
    >
      ground
    </option>
    <option
      value="psychic"
    >
      psychic
    </option>
    <option
      value="rock"
    >
      rock
    </option>
    <option
      value="ice"
    >
      ice
    </option>
    <option
      value="bug"
    >
      bug
    </option>
    <option
      value="dragon"
    >
      dragon
    </option>
    <option
      value="ghost"
    >
      ghost
    </option>
    <option
      value="dark"
    >
      dark
    </option>
    <option
      value="steel"
    >
      steel
    </option>
    <option
      value="fairy"
    >
      fairy
    </option>
  </select>
  Minimum Attack Of:
  <input />
  <button>
    Search
  </button>
  <button
    value="next"
  >
    Next Page
  </button>
  <button
    value="prev"
  >
    Previous Page
  </button>
  <span />
</div>
`;
